src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/StateSynchronizationPerformanceParameters.cs

```diff
@@ -68,75 +83,138 @@ internal class MaterialPropertyPollingFrequency
+            using (StateSynchronizationPerformanceMonitor.Instance.MeasureEventDuration(performanceComponentName, "ShouldUpdateMaterialProperty"))
             {
-                return RenderQueue == PollingFrequency.UpdateContinuously;
-            }
-            else
-            {
-                MaterialPropertyPollingFrequency pollingFrequency;
-                if (PollingFrequencyByMaterialProperty.TryGetValue(new MaterialPropertyKey(materialProperty.ShaderName, materialProperty.propertyName), out pollingFrequency))
+                if (materialProperty.propertyType == MaterialPropertyType.ShaderKeywords)
+                {
+                    return ShaderKeywords == PollingFrequency.UpdateContinuously;
+                }
+                else if (materialProperty.propertyType == MaterialPropertyType.RenderQueue)
+                {
+                    return RenderQueue == PollingFrequency.UpdateContinuously;
+                }
+                else
                 {
-                    switch (pollingFrequency.updateFrequency)
+                    MaterialPropertyPollingFrequency pollingFrequency;
+                    if (PollingFrequencyByMaterialProperty.TryGetValue(new MaterialPropertyKey(materialProperty.ShaderName, materialProperty.propertyName), out pollingFrequency))
                     {
-                        case PollingFrequency.UpdateContinuously:
-                            return true;
-                        case PollingFrequency.UpdateOnceOnStart:
-                            return false;
+                        switch (pollingFrequency.updateFrequency)
+                        {
+                            case PollingFrequency.UpdateContinuously:
+                                return true;
+                            case PollingFrequency.UpdateOnceOnStart:
+                                return false;
+                        }
                     }
-                }
 
-                // If we have a parent, check the parent to see if the parent has an explicit override list
-                if (materialProperties == PollingFrequency.InheritFromParent && parentParameters != null)
-                {
-                    return parentParameters.ShouldUpdateMaterialProperty(materialProperty);
+                    // If we have a parent, check the parent to see if the parent has an explicit override list
+                    if (materialProperties != PollingFrequency.InheritFromParent || parentParameters == null)
+                    {
+                        return materialProperties == PollingFrequency.UpdateContinuously;
+                    }
                 }
+            }
 
-                return materialProperties == PollingFrequency.UpdateContinuously;
+            // Stop the timer before calling parent function
+            return parentParameters.ShouldUpdateMaterialProperty(materialProperty);
+        }
+
+        public bool IsShaderSupported(string shader)

```

Comment:
>IsShaderSupported [](start = 20, length = 17)

Is the point of this method to filter out shaders that have no material properties on them? I guess I don't understand what "supported" means in this case. Maybe a more-descriptive name or doc comment would make this more clear.

Reply:
I want to prevent iterating over all the known properties for each game object when the performance parameters don't support any continuous property synchronization. WIth that in mind, I have changed how this is defined for legibility/what I believe to be a more accurate flag for material synchronization supported.